src/workerd/jsg/modules-new.h

```diff
@@ -0,0 +1,575 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include <workerd/jsg/observer.h>
+#include <workerd/jsg/modules.capnp.h>
+#include <workerd/jsg/url.h>
+#include <workerd/jsg/util.h>
+#include <kj/common.h>
+#include <kj/function.h>
+#include <kj/refcount.h>
+#include <kj/map.h>
+#include <kj/table.h>
+#include <v8.h>
+
+namespace workerd::jsg::modules {
+
+// Defines and implements workerd's (new) module loader subsystem.
```

Comment:
I'm unfamiliar with the old module system, and I think it would be nice to include a sentence or two here about how this new system is different from the old one. 

My impression is that this registry is primarily designed to be more modular and extensible, with the motivation being supporting more diverse specifiers including URL, dynamic imports, and sharing between replicas. Is that correct?